vortex-array/src/view.rs

```diff
@@ -130,26 +130,13 @@ impl ArrayView {
                 Box::leak(Box::new(OpaqueEncoding(child.encoding())))
             });
 
-        // Figure out how many buffers to skip...
-        // We store them depth-first.
-        let buffer_offset = self
-            .flatbuffer()
-            .children()
-            .ok_or_else(|| vortex_err!("flatbuffer children not found"))?
-            .iter()
-            .take(idx)
-            .map(|child| Self::cumulative_nbuffers(child))
-            .sum::<usize>()
-            + self.has_buffer() as usize;
-        let buffer_count = Self::cumulative_nbuffers(child);
-
         Ok(Self {
             encoding,
             dtype: dtype.clone(),
             len,
             flatbuffer: self.flatbuffer.clone(),
             flatbuffer_loc,
-            buffers: self.buffers[buffer_offset..][0..buffer_count].to_vec(),
+            buffers: self.buffers.clone(),
```

Comment:
note that we no longer need to slice, b/c each Array fb message contains the `buffer_index` field, which is a global offset into the buffers list